pkg/storage/pebble_mvcc_scanner.go

```diff
@@ -819,7 +820,13 @@ func (p *pebbleMVCCScanner) getOne(ctx context.Context) (ok, added bool) {
 		if !valid {
 			return false, false
 		}
-		if extended, valid := p.tryDecodeCurrentValueSimple(v); !valid {
+
+		canSkipUncertaintyCheck := !p.checkUncertainty || p.curUnsafeKey.Timestamp.LessEq(p.ts)
+		if !p.mvccHeaderRequired(!canSkipUncertaintyCheck) {
```

Comment:
nit: there's a lot of double negatives here. I'd just replace `canSkipUncertaintyCheck` with `needsUncertaintyCheck` and invert the definition